server/app/services/email/graph/GraphApiEmailClient.java

```diff
@@ -0,0 +1,128 @@
+package services.email.graph;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import com.microsoft.graph.models.BodyType;
+import com.microsoft.graph.models.EmailAddress;
+import com.microsoft.graph.models.ItemBody;
+import com.microsoft.graph.models.Message;
+import com.microsoft.graph.models.Recipient;
+import com.microsoft.graph.users.item.sendmail.SendMailPostRequestBody;
+import com.microsoft.kiota.ApiException;
+import com.typesafe.config.Config;
+import io.prometheus.client.Histogram;
+import java.util.ArrayList;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.Environment;
+import services.cloud.azure.Credentials;
+import services.email.EmailSendClient;
+import services.monitoring.EmailSendMetrics;
+import software.amazon.awssdk.http.HttpStatusCode;
+
+/** GraphApiEmailClient provides methods to send email notifications through Microsoft Graph API. */
+@Singleton
+public class GraphApiEmailClient implements EmailSendClient {
+  public static final String AZURE_SENDER_CONF_PATH = "email.sender";
+  private static final Logger logger = LoggerFactory.getLogger(GraphApiEmailClient.class);
+
+  private final EmailSendMetrics emailSendMetrics;
+  private final GraphApiClientInterface client;
+  private final Environment environment;
+  private final String sender;
+
+  @Inject
+  public GraphApiEmailClient(
+      Credentials credentials,
+      Config config,
+      Environment environment,
+      EmailSendMetrics emailSendMetrics) {
+    this.environment = checkNotNull(environment);
+    this.sender = checkNotNull(config).getString(AZURE_SENDER_CONF_PATH);
```

Comment:
consider check for valid email address?

Reply:
For now, this is the same as the SimpleEmail class, but we can consider doing this later.